specs/form-associated-custom-element.md

```diff
@@ -0,0 +1,114 @@
+- `public readonly validity: ValidityState`
+  - Returns the [validity](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState) of the component
+- `public readonly form: HTMLFormElement | null`
+  - The associated form element
+- `public readonly validationMessage: string`
+  - The current validation message of the element
+- `public readonly labels: Node[]`
+  - Labels associated to the element. See [risks and challenges](#retrieving-labels)
+- `protected abstract proxy: HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement`
+  - A proxy element that will be appended to the DOM when form-association APIs are unavailable. It will server as the custom-elements connection to the form.
+- `protected elementInternals: ElementInternals`
+  - Provides form-association through dedicated APIs.
+- `protected setFormValue(value: File | string | FormData | null, state?: File | string | FormData): void`
+  - When using `elementInternals`, this will set the value in the form. With no form-association support, this will noop because the value will automatically be associated by the proxy element. This can be overridden as necessary by components with more advanced behavior.
+- `protected onLabelClick(): void`: **Discussion**
+  - see [risks and challenges](#clicking-labels)
+- `protected handleKeyPress(): void`
+  - Will submit the form when `enter` key is pressed to match standard input behavior. Can be overridden or extended.
+- `protected setValidity(flags: ValidityStateFlags, message?: string, anchor?: HTMLElement): void;`
+  - With form association support, will set the validity of the component. With no form association, this will noop unless a message is passed. If a message is passed, the proxy element's `setCustomValidity` method will be invoked with the value of the message.
+
+#### Content attributes
+- `id: string`
+  - The id attribute of the component. Used for label association.
+- `value: string | File | FormData`
+  - When provided as a content attribute, value must be a string. When provided as an IDL property, value can be a `string`, `File`, or `FormData`
+- `autofocus: boolean`
+  - Boolean attribute. Will auto-focus the element on page load.
+- `disabled: boolean`
+  - Boolean attribute. Disables the form control, ensuring it doesn't participate in form submission
+- `name: string`
+  - The name of the element. Allows access by name from the associated form.
+- `required: boolean`
+  - Boolean attribute.
+
+### Risks and Challenges
+
+#### Retrieving labels
+Accessing the [`labels`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/labels) property of a native input returns a `NodeList`. In cases where the form-association APIs are available, the implementation from `ElementInternals.labels` works as expected. In cases where we implement a proxy element though, we need to construct the label set from the following:
```

Comment:
I would change this wording to "...In cases where a proxy element is implemented, a label must be constructed from the following:"